### ISSUE TITLE
MPI error on dual processor machine and n>1 processes

### ISSUE THREAD

**Issue by hastingsTheng** [firemodels/fds#10932]
Howdy from Thunderhead!

We've got an international user that is trying to run FDS through Pyrosim using n>1 processes (PyroSim's parallel execution mode), however they encounter the same MPI error on any model they try to run. They are using PyroSim 2022.2 and the bundled FDS6.7.9. A screenshot of the error is attached.

Their basic machine specs:
```
SuperMicro SuperServer
CPU: 2x AMD Epyc 7451 - 24 Core, 48 Threads each, running at 2300MHz 
RAM: 128GB Installed, 115GB available
OS: Win10 Pro 19044
```

![image](https://user-images.githubusercontent.com/87871081/189919352-d0be80c3-c374-49d1-b72d-b8bfeec78932.png)

**Comment by drjfloyd** [comment]
Can they run this outside of Pyrosim using the installer for the current release of FDS on github and following the instructions in the FDS User's Guide?

**Comment by Er9y714** [comment]
I would suggest stopping all MPI Hydra services on Windows Services and run only the latest one.
Secondly. make sure there is only one mpiexec.exe output when **where mpiexec** is run on command line. 


![image](https://user-images.githubusercontent.com/26166281/189926025-d9a1942e-873a-41f3-b2b6-62ddb0aa62b1.png)

**Comment by hastingsTheng** [comment]
@drjfloyd @Er9y714 Thanks for the quick replies. I have reached back out to them with some steps to try both of the above and will reply again when I hear back from them.

**Comment by hastingsTheng** [comment]
@drjfloyd The user is not able to run with the current release. Running with the standalone version of FDS results in the same MPI error.
@Er9y714 The user stopped all Hydra services except 2021.6 and still ran in to the same issue.

**Comment by mcgratta** [comment]
Try this command
```
fds_local -F shm jobname.fds
```
or
```
fds_local -F shm:ofi jobname.fds
```
We had some issues a while ago related to AMD processors and MPI communication fabrics.

**Comment by drjfloyd** [comment]
If Kevin's suggestion doesn't work might be good at that point to rule out non-MPI causes. Has this user been able to get this case or another case that failed with MPI to start time stepping when running without MPI?

**Comment by hastingsTheng** [comment]
Thanks Kevin. I think I ran across the issue referencing the shm fix before opening this case. I'll have them try that.

@drjfloyd Yes, they are able to run their cases without MPI just fine. It's only when they try to use MPI that they run in to any issues.

**Comment by hastingsTheng** [comment]
@mcgratta @drjfloyd Unfortunately the shm argument did not fix this issue. The user ran in to the same MPI error with that as they did without.

**Comment by mcgratta** [comment]
Maybe you have tried this already, but try this. Open the CMDfds command prompt window and type
```
mpiexec -n 14 -localonly test_mpi
```
This is just a "Hello World" program and does not involve FDS at all. It will just test your installation of the Hydra process manager.

**Comment by hastingsTheng** [comment]
@mcgratta Hadn't tried that, but the user gets the same mpi error.

**Comment by hastingsTheng** [comment]
@mcgratta Could this be a larger issue with MPI? We have seen an increased number of these types of MPI errors in the past 2 or 3 weeks, mainly from our international users. Something related to system locale and special characters perhaps?

**Comment by mcgratta** [comment]
It's usually a case of missing libraries for a particular flavor of Windows. The AMD processor issue complicates. Try this
```
mpiexec -n 14 hostname
```

**Comment by mcgratta** [comment]
Did this last test work? This test has nothing to do with FDS. It only tests whether the process manager is running.

**Comment by hastingsTheng** [comment]
@mcgratta Sorry, I hadn't seen your last comment till just now. I'll pass that along to the user and have them give it a shot.

**Comment by hastingsTheng** [comment]
@mcgratta Just heard back. They encountered the same mpi error.

**Comment by mcgratta** [comment]
I am out of ideas. This person would have to install the Intel oneAPI compilers and libraries and compile the code. This is not easy to do. I cannot think of what libraries are currently missing.

**Comment by cunconbkhp** [comment]
Hi, i would like to know if the problem has been solved? my team are testing a dual core AMD machine (2*32 procs). I've installed FDS 6.9 and we could not used more than 12 processor (mpiexec -n 12 ) for our simulation in cmdFDS.  The errors lines are like below : 

[mpiexec@WIN-JTLRBJE3MVD] wmain (mpiexec.c:2165): assert (pg->intel.exitcodes != NULL) failed
[mpiexec@WIN-JTLRBJE3MVD] HYD_sock_write (..\windows\src\hydra_sock.c:387): write error (errno = 34)

Thanks!

**Comment by mcgratta** [comment]
Run this command in the CMDfds prompt:
```
mpiexec -n 2 test_mpi
```
If that is successful, run this
```
mpiexec -n 20 test_mpi
```
Post the output in this thread, like this:

![image](https://github.com/user-attachments/assets/e7548e92-f269-49d0-8592-41726334aa6b)

**Comment by cunconbkhp** [comment]
Hi Kevin, thanks for your answer.

In fact i've checked that for sum number it does not work. For any numbre from 1 to 21 and 33 to 64 no problem, but from 22 to 32, the error appears.   
![Capture_error](https://github.com/user-attachments/assets/bdcdacf5-cfca-43d4-91b2-ec94393815fc)

**Comment by mcgratta** [comment]
Open up the Task Manager and show me this

![image](https://github.com/user-attachments/assets/7f96bd34-c663-4110-aa5a-88bec43374e8)

**Comment by cunconbkhp** [comment]
Hi Kevin, here it is : 
![Capture_task](https://github.com/user-attachments/assets/cc154dd5-54e6-4fe4-9b60-d900fae7b0d7)

I'm running two simulations, one with 36 procs and another with 12 procs.

**Comment by mcgratta** [comment]
Are you able to view [this thread](https://community.intel.com/t5/Intel-MPI-Library/MPI-Hydra-and-pinning-issue-on-dual-socket-AMD/m-p/1509597#M10825) ? It seems relevant to your situation.

**Comment by withnoempty** [comment]
My computer uses the Windows operating system, and the processor is AMD EPYC 7R32 48-Core Processor with a frequency of 2.80GHz and there are 2 processors. I also encountered the same error. It can calculate normally when MPI is not required, but an error will be reported during parallel computing. It is worth noting that at the beginning, when performing single-core calculations, the computer indicated that vcruntime.dll was missing. After that, I downloaded Visual C++ Redistributable X64, and then I could perform single-core simulations.
